tests/test_app/components.py

```diff
@@ -135,6 +136,67 @@ def django_css():
     )
 
 
+@component
+def django_css_allow_duplicates():
+    scope = reactpy_django.hooks.use_scope()
+    css_files, set_css_files = hooks.use_state(
+        [
+            reactpy_django.components.django_css(
+                "django-css-only-once-test.css", allow_duplicates=True
+            )
+        ]
+    )
+
+    async def add_end_css(event):
+        set_css_files(
+            css_files
+            + [
+                reactpy_django.components.django_css(
+                    "django-css-only-once-test.css",
+                    allow_duplicates=True,
+                    key=str(uuid4()),
+                )
+            ]
+        )
+
+    async def add_front_css(event):
+        set_css_files(
+            [
+                reactpy_django.components.django_css(
+                    "django-css-only-once-test.css",
+                    allow_duplicates=True,
+                    key=str(uuid4()),
+                )
+            ]
+            + css_files
+        )
```

Comment:
@rmorshea Is changing a list supposed to cause all components in the list to re-render every time?

I know the identity of the `list` is changing, but it's still a list. Not sure if you intended for swapping out lists to cause everything to unmount and re-render.

Reply:
Did some more debugging and I've managed to characterize this issue. I forgot that `use_effect(..., dependencies=None)` cleans up after every render. I [created an issue](https://github.com/reactive-python/reactpy/issues/1197) to see if there's viability in an alternative effect condition.

I tried creating a workaround, but was unsuccessful in getting `use_effect(..., dependencies=[])` unmount to trigger when the component gets removed from a list. This might actually be a significant bug.

```python
@component
def _django_static_file(
    static_path: str,
    prevent_duplicates: bool,
    file_type: str,
    vdom_constructor: VdomDictConstructor,
):
    scope = use_scope()
    mount_trigger, set_mount_trigger = hooks.use_state(True)
    ownership_uuid = hooks.use_memo(lambda: uuid4())

    # Configure the ASGI scope to track the file
    if prevent_duplicates:
        scope.setdefault("reactpy", {}).setdefault(file_type, {})
        scope["reactpy"][file_type].setdefault(static_path, ownership_uuid)

    # Check if other _django_static_file components have unmounted
    @hooks.use_effect(dependencies=None)
    async def mount_manager():
        if prevent_duplicates and not scope["reactpy"][file_type].get(static_path):
            scope["reactpy"][file_type].setdefault(static_path, ownership_uuid)
            set_mount_trigger(not mount_trigger)

    # Notify other components that we've unmounted
    @hooks.use_effect(dependencies=[])
    async def unmount_manager():
        if not prevent_duplicates:
            return

        def unmount():
            if scope["reactpy"][file_type].get(static_path) == ownership_uuid:
                scope["reactpy"][file_type].pop(static_path)

        return unmount

    # Render the component, if needed
    if not prevent_duplicates or (
        scope["reactpy"][file_type].get(static_path) == ownership_uuid
    ):
        return vdom_constructor(cached_static_contents(static_path))
```